halpybot/commands/caseutils.py

```diff
@@ -58,41 +97,555 @@ async def cmd_go(ctx: Context, args: List[str]):
     )
 
 
+# BOARD AND CASE LISTING
 @Commands.command("listboard")
+@Require.permission(Drilled)
 async def cmd_listboard(ctx: Context, args: List[str]):
     """
     Send a user the key details of every case on the board in DMs
 
-    Usage: !listboard
+    Usage: !listboard [Platform or Case Type]
     Aliases: n/a
     """
+    # Get the Case Board
     caseboard = ctx.bot.board.by_id
     if not caseboard:
         return await ctx.redirect("The case board is empty!")
-    message = "Here's the current case board:\n"
-    for case in caseboard.values():
-        hskf = "Seal" if case.hull_percent else "Fisher" if case.planet else "Unknown"
+    # Process Args (If Exist)
+    if args:
+        platforms = [
+            member.name.casefold().replace("_horizons", "") for member in Platform
+        ]
+        casetypes = [member.name.casefold() for member in CaseType]
+        list_filter = args[0].casefold()
+        if list_filter not in casetypes and list_filter not in platforms:
+            list_filter = None
+    else:
+        list_filter = None
+
+    filtered_cases = [
```

Comment:
[optimization] replace the list comprehension with a series of generator expressions.
This will allow you to reduce the cognitive complexity and increase expressiveness.
It also turns your current 2 * O(n) into an O(n)